develop: TestWindow_DragIntoNewObjectKeepingFocus & TestWindow_HoverableOnDragging fail

### Describe the bug:
Windows10 and a linux Mint 20 VM are both failing with the following errors
--- FAIL: TestWindow_DragIntoNewObjectKeepingFocus (0.88s)
    window_test.go:306: 
        	Error Trace:	window_test.go:306
        	Error:      	Not equal: 
        	            	expected: *desktop.MouseEvent(&desktop.MouseEvent{PointEvent:fyne.PointEvent{AbsolutePosition:fyne.Position{X:19, Y:9}, Position:fyne.Position{X:15, Y:5}}, Button:1, Modifier:0})
        	            	actual  : <nil>(<nil>)
        	Test:       	TestWindow_DragIntoNewObjectKeepingFocus
    window_test.go:316: 
        	Error Trace:	window_test.go:316
        	Error:      	Expected nil, but got: &desktop.MouseEvent{PointEvent:fyne.PointEvent{AbsolutePosition:fyne.Position{X:19, Y:9}, Position:fyne.Position{X:1, Y:5}}, Button:1, Modifier:0}
        	Test:       	TestWindow_DragIntoNewObjectKeepingFocus
FAIL
--- FAIL: TestWindow_HoverableOnDragging (0.96s)
    window_test.go:377: 
        	Error Trace:	window_test.go:377
        	Error:      	Expected nil, but got: &desktop.MouseEvent{PointEvent:fyne.PointEvent{AbsolutePosition:fyne.Position{X:8, Y:8}, Position:fyne.Position{X:4, Y:4}}, Button:1, Modifier:0}
        	Test:       	TestWindow_HoverableOnDragging
FAIL

Started with commit
commit bb4c70a554d62478e6897e8fbc2368eb47c3d834
Author: Andy Williams <andy@andywilliams.me>
Date:   Wed Jan 13 20:21:17 2021 +0000

    move drag init into mouse moved


### To Reproduce:
go test ./... -count=1

### Device (please complete the following information):
 - **OS:** Linux and Windows
 - **Version:** Mint 20 and Windows 10
 - **Go version:** 1.15.6
 - **Fyne version:** develop HEAD

--- a/internal/driver/glfw/window.go
+++ b/internal/driver/glfw/window.go
@@ -621,6 +621,21 @@ func (w *window) mouseMoved(viewport *glfw.Window, xpos float64, ypos float64) {
 			w.customCursor = rawCursor
 		}
 	}
+
+	if w.mouseButton != 0 && !w.mouseDragStarted {
+		obj, pos, _ := w.findObjectAtPositionMatching(w.canvas, previousPos, func(object fyne.CanvasObject) bool {
+			_, ok := object.(fyne.Draggable)
+			return ok
+		})
+
+		if wid, ok := obj.(fyne.Draggable); ok {
+			w.mouseDragPos = previousPos
+			w.mouseDragged = wid
+			w.mouseDraggedOffset = previousPos.Subtract(pos)
+			w.mouseDragStarted = true
+		}
+	}
+
 	if obj != nil && !w.objIsDragged(obj) {
 		ev := new(desktop.MouseEvent)
 		ev.AbsolutePosition = w.mousePos
@@ -637,14 +652,6 @@ func (w *window) mouseMoved(viewport *glfw.Window, xpos float64, ypos float64) {
 		}
 	} else if w.mouseOver != nil && !w.objIsDragged(w.mouseOver) {
 		w.mouseOut()
-	}
-
-	if wid, ok := obj.(fyne.Draggable); ok {
-		if w.mouseButton != 0 && !w.mouseDragStarted {
-			w.mouseDragPos = previousPos
-			w.mouseDragged = wid
-			w.mouseDraggedOffset = w.mousePos.Subtract(pos)
-		}
 	}
 
 	if w.mouseDragged != nil {
@@ -716,7 +723,14 @@ func (w *window) mouseClicked(_ *glfw.Window, btn glfw.MouseButton, action glfw.
 		if action == glfw.Press {
 			w.queueEvent(func() { wid.MouseDown(mev) })
 		} else if action == glfw.Release {
-			w.queueEvent(func() { wid.MouseUp(mev) })
+			dragged := w.mouseDragged.(interface{}).(fyne.CanvasObject)
+			_, tappableDrag := dragged.(desktop.Mouseable)
+			if dragged != nil && tappableDrag {
+				mev.Position = w.mousePos.Subtract(w.mouseDraggedOffset)
+				w.queueEvent(func() { dragged.(desktop.Mouseable).MouseUp(mev) })
+			} else {
+				w.queueEvent(func() { wid.MouseUp(mev) })
+			}
 		}
 	}
 

